pipit/trace.py

```diff
@@ -62,180 +71,227 @@ def _match_events(self):
+                )
 
-                    event_types = list(filtered_df["Event Type"])
-                    df_indices, timestamps = list(filtered_df.index), list(
-                        filtered_df["Timestamp (ns)"]
+                # Iterate through all events of filtered DataFrame
+                for i in range(len(filtered_df)):
+                    curr_df_index, curr_timestamp, evt_type = (
+                        df_indices[i],
+                        timestamps[i],
+                        event_types[i],
                     )
 
-                    # Iterate through all events of filtered DataFrame
-                    for i in range(len(filtered_df)):
-                        curr_df_index, curr_timestamp, evt_type = (
-                            df_indices[i],
-                            timestamps[i],
-                            event_types[i],
-                        )
-
-                        if evt_type == "Enter":
-                            # Add current dataframe index and timestamp to stack
-                            stack.append((curr_df_index, curr_timestamp))
-                        else:
-                            # Pop corresponding enter event's dataframe index
-                            # and timestamp
-                            enter_df_index, enter_timestamp = stack.pop()
-
-                            # Fill in the lists with the matching values
-                            matching_events[enter_df_index] = curr_df_index
-                            matching_events[curr_df_index] = enter_df_index
-
-                            matching_times[enter_df_index] = curr_timestamp
-                            matching_times[curr_df_index] = enter_timestamp
+                    if evt_type == "Enter":
+                        # Add current dataframe index and timestamp to stack
+                        stack.append((curr_df_index, curr_timestamp))
+                    else:
+                        # Pop corresponding enter event's dataframe index
+                        # and timestamp
+                        enter_df_index, enter_timestamp = stack.pop()
+
+                        # Fill in the lists with the matching values
+                        matching_events[enter_df_index] = curr_df_index
+                        matching_events[curr_df_index] = enter_df_index
+
+                        matching_times[enter_df_index] = curr_timestamp
+                        matching_times[curr_df_index] = enter_timestamp
 
             self.events["_matching_event"] = matching_events
             self.events["_matching_timestamp"] = matching_times
 
             self.events = self.events.astype({"_matching_event": "Int32"})
 
     def _match_caller_callee(self):
-        """Matches callers (parents) to callees (children) and adds three
+        """Matches callers (parents) to callees (children) and adds two
         columns to the dataframe:
-        _depth, _parent, and _children
```

Comment:
removed depth column generated by this function as it will be stored in the cct node